mlflow/server/js/src/experiment-tracking/components/CompareRunArtifactView.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from 'react';
+import ShowArtifactPage from './artifact-view-components/ShowArtifactPage';
+import { RunInfoEntity } from '../types';
+import { useRunsArtifacts } from './experiment-page/hooks/useRunsArtifacts';
+import { getCommonArtifacts } from './experiment-page/utils/getCommonArtifacts';
+import { useDesignSystemTheme } from '@databricks/design-system';
+import { FormattedMessage } from 'react-intl';
+import { ArtifactViewTree } from './ArtifactViewTree';
+import { getBasename } from '../../common/utils/FileUtils';
+
+export const CompareRunArtifactView = ({
+  runUuids,
+  runInfos,
+  colWidth,
+}: {
+  runUuids: string[];
+  runInfos: RunInfoEntity[];
+  colWidth: number;
+}) => {
+  const { theme } = useDesignSystemTheme();
+  const [artifactPath, setArtifactPath] = useState<string | undefined>();
+
+  const { artifactsKeyedByRun } = useRunsArtifacts(runUuids);
+  const commonArtifacts = getCommonArtifacts(artifactsKeyedByRun);
+
+  if (commonArtifacts.length === 0) {
+    return (
+      <h2>
+        <FormattedMessage
+          defaultMessage="No common artifacts to display."
+          description="Text shown when there are no common artifacts between the runs"
+        />
+      </h2>
+    );
+  }
+  return (
+    <div
+      css={{
+        display: 'flex',
+        flexDirection: 'row',
+        height: '100vh',
+      }}
+    >
+      <div
+        css={{
+          backgroundColor: theme.colors.backgroundPrimary,
+          color: theme.colors.textPrimary,
+          flex: '1 1 0%',
+          whiteSpace: 'nowrap',
+          border: `1px solid ${theme.colors.grey300}`,
+          overflowY: 'auto',
+        }}
+      >
+        <ArtifactViewTree
+          data={commonArtifacts.map((path: string) => ({
+            id: path,
+            active: artifactPath === path,
+            name: getBasename(path),
+          }))}
```

Comment:
looks like this is giving a typescript error due to the `Record<string, string>[]` type of `data`:

https://github.com/mlflow/mlflow/blob/82e944620080bb48a0e944b17966f6abf193a1ee/mlflow/server/js/src/experiment-tracking/components/ArtifactViewTree.tsx#L19

this type is actually incorrect, and it should be like this (ref [Treebeard](https://www.npmjs.com/package/react-treebeard) docs):

```ts
type TreebeardData = {
    id?: string,
    name: string,
    children?: TreebeardData[],
    toggled?: boolean,
    active?: boolean,
    loading?: boolean,
    decorators?: any,
    animations?: any
}
```

if you really wanted to, you could fix the type, but since you're not the one that wrote it, i'm okay with just typing this object as `any`:

```suggestion
          data={commonArtifacts.map((path: string) => ({
            id: path,
            active: artifactPath === path,
            name: getBasename(path),
          })) as any}
```

Reply:
I decided to fix it directly on the `ArtifactViewTree`!